Client/Application/Migration.swift

```diff
@@ -22,13 +29,28 @@ class Migration {
         }
         
         // `.migrate` is called in `BrowserViewController.viewDidLoad()`
-        if !Preferences.Chromium.syncV2BookmarksMigrationCompleted.value {
-            braveCoreBookmarksMigrator = BraveCoreMigrator()
+        if !isChromiumMigrationCompleted {
+            braveCoreSyncObjectsMigrator = BraveCoreMigrator()
         }
         
         if !Preferences.Migration.playlistV1FileSettingsLocationCompleted.value {
             movePlaylistV1Items()
         }
+        
+        // Adding Observer to enable sync types 
+        NotificationCenter.default.addObserver(
```

Comment:
Maybe we can add the observer that uses `Publisher` instead? So we know for sure it is deleted  when the `deinit` is called.

Reply:
Oh we should certainly get rid of the notification center and use combine here. If it is alright I would like to do this while we are handling the Observer API changes in core side with this ticket.
https://github.com/brave/brave-ios/issues/3757